src/roswire/proxy/roslaunch/ros2launch.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+__all__ = ('ROS2LaunchManager',)
+
+from typing import Collection, List, Mapping, Optional, Sequence, Tuple, Union
+import os
+import shlex
+
+from loguru import logger
+import attr
+import dockerblade
+
+from .config import LaunchConfig
+from .controller import ROSLaunchController
+from ... import exceptions as exc
+from .app.app import App
+
+
+@attr.s(eq=False)
+class ROS2LaunchManager:
+    """Provides access to `roslaunch <wiki.ros.org/roslaunch/>`_ for an
+    associated ROS system. This interface is used to locate, read, and write
+    `launch XML files <http://wiki.ros.org/roslaunch/XML>`_,
+    and to launch ROS nodes using those files.
+    """
+    _shell: dockerblade.shell.Shell = attr.ib(repr=False)
+    _files: dockerblade.files.FileSystem = attr.ib(repr=False)
+
```

Comment:
Let's add
```
@classmethod
def for_app_instance(cls, app_instance: 'AppInstance') -> 'ROS2LaunchManager`:
    return ROS2LaunchManager(....)
```

That way we're robust to any changes in the constructor :-) (Btw, if you're curious about these sorts of API choices in general, you should check out Effective Java by Joshua Block. The book is for Java, but many of the same lessons can be applied to other object-oriented languages.)